src/bootstrap/src/core/builder/mod.rs

```diff
@@ -1212,9 +1206,21 @@ impl<'a> Builder<'a> {
                 sysroot
             }
         }
+
         self.ensure(Libdir { compiler, target })
     }
 
+    /// Returns the bindir for a compiler's sysroot.
+    pub fn sysroot_target_bindir(&self, compiler: Compiler, target: TargetSelection) -> PathBuf {
+        self.ensure_sysroot_target_dir(compiler, target).join(target).join("bin")
+    }
+
+    /// Returns the libdir where the standard library and other artifacts are
+    /// found for a compiler's sysroot.
+    pub fn sysroot_target_libdir(&self, compiler: Compiler, target: TargetSelection) -> PathBuf {
+        self.ensure_sysroot_target_dir(compiler, target).join(target).join("lib")
+    }
```

Comment:
Remark [COM 3/3]: ... so that `sysroot_target_bindir` can construct a path with child directories without having to backtrack, and they now clearly look related.